### ISSUE TITLE
si.usb(cb) does not work

### ISSUE THREAD

**Issue by Zoliex** [sebhildebrandt/systeminformation#645]
**Describe the bug**
Hello yesterday i have coded this on Windows 10 family edition :

`require('systeminformation').usb().then(json => {
    console.log(JSON.stringify(json));
});`

But when I run my code, I always get this callback :
> []

I have a Wi-Fi key connected to my USB ports and a mouse.
Why the software can't detect my USB peripherals ?
(Please excuse me for my English, I'm French)

**Comment by Zoliex** [comment]
When i run this command, i get this:
Get-WmiObject CIM_LogicalDevice             

Manufacturer                     Name
------------                     ----
Realtek Semiconductor Corp.      TP-Link Wireless USB Adapte... 
+ [MOUSE]

and when I run this command, that is the command run by your plugin I get nothing : 
Get-WmiObject CIM_LogicalDevice | where { $_.Name -match "^USB"}

**Comment by sebhildebrandt** [comment]
@Zoliex ... this is strange, you should get much more information such as all USB Hubs, Host controllers etc. On my machines - except of things like mouse and keyboard - I get at least 6 entries ... here an example:

```
...
__GENUS                     : 2
__CLASS                     : Win32_PnPEntity
__SUPERCLASS                : CIM_LogicalDevice
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_PnPEntity.DeviceID="PCI\\VEN_1B36&DEV_000D&SUBSYS_11001AF4&REV_01\\3&11583659&0&20"
__PROPERTY_COUNT            : 26
__DERIVATION                : {CIM_LogicalDevice, CIM_LogicalElement, CIM_ManagedSystemElement}
__SERVER                    : DESKTOP-8P7EIGH
__NAMESPACE                 : root\cimv2
__PATH                      : \\DESKTOP-8P7EIGH\root\cimv2:Win32_PnPEntity.DeviceID="PCI\\VEN_1B36&DEV_000D&SUBSYS_11001AF4&REV_01\\3&11583659&0&20"
Availability                :
Caption                     : Standard USB 3.0 eXtensible Host Controller - 1.0 (Microsoft)
ClassGuid                   : {36fc9e60-c465-11cf-8056-444553540000}
CompatibleID                : {PCI\VEN_1B36&DEV_000D&REV_01, PCI\VEN_1B36&DEV_000D, PCI\VEN_1B36&CC_0C0330, PCI\VEN_1B36&CC_0C03...}
ConfigManagerErrorCode      : 0
ConfigManagerUserConfig     : False
CreationClassName           : Win32_PnPEntity
Description                 : USB xHCI Compliant Host Controller
DeviceID                    : PCI\VEN_1B36&DEV_000D&SUBSYS_11001AF4&REV_01\3&11583659&0&20
ErrorCleared                :
ErrorDescription            :
HardwareID                  : {PCI\VEN_1B36&DEV_000D&SUBSYS_11001AF4&REV_01, PCI\VEN_1B36&DEV_000D&SUBSYS_11001AF4, PCI\VEN_1B36&DEV_000D&CC_0C0330,
                              PCI\VEN_1B36&DEV_000D&CC_0C03}
InstallDate                 :
LastErrorCode               :
Manufacturer                : Generic USB xHCI Host Controller
Name                        : Standard USB 3.0 eXtensible Host Controller - 1.0 (Microsoft)
PNPClass                    : USB
PNPDeviceID                 : PCI\VEN_1B36&DEV_000D&SUBSYS_11001AF4&REV_01\3&11583659&0&20
PowerManagementCapabilities :
PowerManagementSupported    :
Present                     : True
Service                     : USBXHCI
Status                      : OK
StatusInfo                  :
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : DESKTOP-8P7EIGH
PSComputerName              : DESKTOP-8P7EIGH
...
```

Can you confirm that you do not get any of those?? Then this might be an issue that I am not able to reproduce ... 

It would be great if you can provide FULL output of 

```
Get-WmiObject CIM_LogicalDevice | where { $_.Description -match "^USB"}
```

as well as 

```
Get-WmiObject CIM_LogicalDevice  | fl *
```

**Comment by Zoliex** [comment]
When I run ```Get-WmiObject CIM_LogicalDevice | where { $_.Description -match "^USB"}``` I get nothing and when I run ```Get-WmiObject CIM_LogicalDevice  | fl * > command.txt

``` I get a very long log ([command.txt](https://github.com/sebhildebrandt/systeminformation/files/7911909/command.txt))

**Comment by Zoliex** [comment]
At line 3392 of command.txt, I can see that my USB adapter is detected :
```
PSComputerName              : LAPTOP-D47E8127
Availability                : 3
Name                        : TP-Link Wireless USB Adapter #4
Status                      : 
StatusInfo                  : 
DeviceID                    : 2
__GENUS                     : 2
__CLASS                     : Win32_NetworkAdapter
__SUPERCLASS                : CIM_NetworkAdapter
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_NetworkAdapter.DeviceID="2"
__PROPERTY_COUNT            : 40
__DERIVATION                : {CIM_NetworkAdapter, CIM_LogicalDevice, CIM_LogicalElement, CIM_ManagedSystemElement}
__SERVER                    : LAPTOP-D47E8127
__NAMESPACE                 : root\cimv2
__PATH                      : \\LAPTOP-D47E8127\root\cimv2:Win32_NetworkAdapter.DeviceID="2"
AdapterType                 : Ethernet 802.3
AdapterTypeId               : 0
AutoSense                   : 
Caption                     : [00000002] TP-Link Wireless USB Adapter
ConfigManagerErrorCode      : 0
ConfigManagerUserConfig     : False
CreationClassName           : Win32_NetworkAdapter
Description                 : TP-Link Wireless USB Adapter
ErrorCleared                : 
ErrorDescription            : 
GUID                        : {B0898476-6449-4A94-89F4-D91559191250}
Index                       : 2
InstallDate                 : 
Installed                   : True
InterfaceIndex              : 25
LastErrorCode               : 
MACAddress                  : 28:EE:52:09:DB:A7
Manufacturer                : Realtek Semiconductor Corp.
MaxNumberControlled         : 0
MaxSpeed                    : 
NetConnectionID             : Wi-Fi 7
NetConnectionStatus         : 2
NetEnabled                  : True
NetworkAddresses            : 
PermanentAddress            : 
PhysicalAdapter             : True
PNPDeviceID                 : USB\VID_2357&PID_012D\123456
PowerManagementCapabilities : 
PowerManagementSupported    : False
ProductName                 : TP-Link Wireless USB Adapter
ServiceName                 : RtlWlanu
Speed                       : 144400000
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : LAPTOP-D47E8127
TimeOfLastReset             : 20220119205213.500676+060
Scope                       : System.Management.ManagementScope
Path                        : \\LAPTOP-D47E8127\root\cimv2:Win32_NetworkAdapter.DeviceID="2"
Options                     : System.Management.ObjectGetOptions
ClassPath                   : \\LAPTOP-D47E8127\root\cimv2:Win32_NetworkAdapter
Properties                  : {AdapterType, AdapterTypeId, AutoSense, Availability...}
SystemProperties            : {__GENUS, __CLASS, __SUPERCLASS, __DYNASTY...}
Qualifiers                  : {dynamic, Locale, provider, UUID}
Site                        : 
Container                   : 

```

**Comment by Zoliex** [comment]
And when I run this command (old command without ^ at -match "^USB"): 
```
Get-WmiObject CIM_LogicalDevice | where { $_.Description -match "USB"}
```
I get all my USB devices : 
```
__GENUS                     : 2
__CLASS                     : Win32_PnPEntity
__SUPERCLASS                : CIM_LogicalDevice
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_PnPEntity.DeviceID="USB\\VID_2357&PID_012D\\123456"
__PROPERTY_COUNT            : 26
__DERIVATION                : {CIM_LogicalDevice, CIM_LogicalElement, CIM_ManagedSystemElement}
__SERVER                    : LAPTOP-D47E8127
__NAMESPACE                 : root\cimv2
__PATH                      : \\LAPTOP-D47E8127\root\cimv2:Win32_PnPEntity.DeviceID="USB\\VID_2357&PID_012D\\123456"
Availability                :
Caption                     : TP-Link Wireless USB Adapter #4
ClassGuid                   : {4d36e972-e325-11ce-bfc1-08002be10318}
CompatibleID                : {USB\Class_FF&SubClass_FF&Prot_FF, USB\Class_FF&SubClass_FF, USB\Class_FF}
ConfigManagerErrorCode      : 0
ConfigManagerUserConfig     : False
CreationClassName           : Win32_PnPEntity
Description                 : Adaptateur USB TP-Link sans fil
DeviceID                    : USB\VID_2357&PID_012D\123456
ErrorCleared                :
ErrorDescription            :
HardwareID                  : {USB\VID_2357&PID_012D&REV_0210, USB\VID_2357&PID_012D}
InstallDate                 : 
LastErrorCode               :
Manufacturer                : Realtek Semiconductor Corp.
Name                        : TP-Link Wireless USB Adapter #4
PNPClass                    : Net
PNPDeviceID                 : USB\VID_2357&PID_012D\123456
PowerManagementCapabilities :
PowerManagementSupported    :
Present                     : True
Service                     : RtlWlanu
Status                      : OK
StatusInfo                  :
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : LAPTOP-D47E8127
PSComputerName              : LAPTOP-D47E8127

__GENUS                     : 2
__CLASS                     : Win32_PnPEntity
__SUPERCLASS                : CIM_LogicalDevice
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_PnPEntity.DeviceID="USB\\VID_04F2&PID_B571\\5&651EB89&0&7"
__PROPERTY_COUNT            : 26
__DERIVATION                : {CIM_LogicalDevice, CIM_LogicalElement, CIM_ManagedSystemElement}
__SERVER                    : LAPTOP-D47E8127
__NAMESPACE                 : root\cimv2
__PATH                      : \\LAPTOP-D47E8127\root\cimv2:Win32_PnPEntity.DeviceID="USB\\VID_04F2&PID_B571\\5&651EB89&0&7"
Availability                :
Caption                     : Périphérique USB composite
ClassGuid                   : {36fc9e60-c465-11cf-8056-444553540000}
CompatibleID                : {USB\DevClass_00&SubClass_00&Prot_00, USB\DevClass_00&SubClass_00, USB\DevClass_00, USB\COMPOSITE}
ConfigManagerErrorCode      : 0
ConfigManagerUserConfig     : False
CreationClassName           : Win32_PnPEntity
Description                 : Périphérique USB composite
DeviceID                    : USB\VID_04F2&PID_B571\5&651EB89&0&7
ErrorCleared                :
ErrorDescription            :
HardwareID                  : {USB\VID_04F2&PID_B571&REV_9952, USB\VID_04F2&PID_B571}
InstallDate                 :
LastErrorCode               :
Manufacturer                : (Contrôleur hôte USB standard)
Name                        : Périphérique USB composite
PNPClass                    : USB
PNPDeviceID                 : USB\VID_04F2&PID_B571\5&651EB89&0&7
PowerManagementCapabilities :
PowerManagementSupported    :
Present                     : True
Service                     : usbccgp
Status                      : OK
StatusInfo                  :
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : LAPTOP-D47E8127
PSComputerName              : LAPTOP-D47E8127

__GENUS                     : 2
__CLASS                     : Win32_PnPEntity
__SUPERCLASS                : CIM_LogicalDevice
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_PnPEntity.DeviceID="PCI\\VEN_8086&DEV_9D2F&SUBSYS_10941025&REV_21\\3&11583659&0&A0"
__PROPERTY_COUNT            : 26
__DERIVATION                : {CIM_LogicalDevice, CIM_LogicalElement, CIM_ManagedSystemElement}
__SERVER                    : LAPTOP-D47E8127
__NAMESPACE                 : root\cimv2
__PATH                      : \\LAPTOP-D47E8127\root\cimv2:Win32_PnPEntity.DeviceID="PCI\\VEN_8086&DEV_9D2F&SUBSYS_10941025&REV_21\\3&11583659&0&A0"    
Availability                : 
Caption                     : Contrôleur hôte Intel(R) USB 3.0 eXtensible - 1.0 (Microsoft)
ClassGuid                   : {36fc9e60-c465-11cf-8056-444553540000}
CompatibleID                : {PCI\VEN_8086&DEV_9D2F&REV_21, PCI\VEN_8086&DEV_9D2F, PCI\VEN_8086&CC_0C0330, PCI\VEN_8086&CC_0C03...}
ConfigManagerErrorCode      : 0
ConfigManagerUserConfig     : False
CreationClassName           : Win32_PnPEntity
Description                 : Contrôleur d’hôte compatible xHCI USB
DeviceID                    : PCI\VEN_8086&DEV_9D2F&SUBSYS_10941025&REV_21\3&11583659&0&A0
ErrorCleared                :
ErrorDescription            :
HardwareID                  : {PCI\VEN_8086&DEV_9D2F&SUBSYS_10941025&REV_21, PCI\VEN_8086&DEV_9D2F&SUBSYS_10941025, PCI\VEN_8086&DEV_9D2F&CC_0C0330,    
                              PCI\VEN_8086&DEV_9D2F&CC_0C03}
InstallDate                 :
LastErrorCode               : 
Manufacturer                : Contrôleur d’hôte xHCI USB générique
Name                        : Contrôleur hôte Intel(R) USB 3.0 eXtensible - 1.0 (Microsoft)
PNPClass                    : USB
PNPDeviceID                 : PCI\VEN_8086&DEV_9D2F&SUBSYS_10941025&REV_21\3&11583659&0&A0
PowerManagementCapabilities :
PowerManagementSupported    :
Present                     : True
Service                     : USBXHCI
Status                      : OK
StatusInfo                  :
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : LAPTOP-D47E8127
PSComputerName              : LAPTOP-D47E8127

__GENUS                     : 2
__CLASS                     : Win32_PnPEntity
__SUPERCLASS                : CIM_LogicalDevice
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_PnPEntity.DeviceID="USB\\ROOT_HUB30\\4&5CDA812&0&0"
__PROPERTY_COUNT            : 26
__DERIVATION                : {CIM_LogicalDevice, CIM_LogicalElement, CIM_ManagedSystemElement}
__SERVER                    : LAPTOP-D47E8127
__NAMESPACE                 : root\cimv2
__PATH                      : \\LAPTOP-D47E8127\root\cimv2:Win32_PnPEntity.DeviceID="USB\\ROOT_HUB30\\4&5CDA812&0&0"
Availability                :
Caption                     : Hub USB racine (USB 3.0)
ClassGuid                   : {36fc9e60-c465-11cf-8056-444553540000}
CompatibleID                :
ConfigManagerErrorCode      : 0
ConfigManagerUserConfig     : False
CreationClassName           : Win32_PnPEntity
Description                 : Hub USB racine (USB 3.0)
DeviceID                    : USB\ROOT_HUB30\4&5CDA812&0&0
ErrorCleared                :
ErrorDescription            :
HardwareID                  : {USB\ROOT_HUB30&VID8086&PID9D2F&REV0021, USB\ROOT_HUB30&VID8086&PID9D2F, USB\ROOT_HUB30}
InstallDate                 :
LastErrorCode               :
Manufacturer                : (HUB USB standard)
Name                        : Hub USB racine (USB 3.0)
PNPClass                    : USB
PNPDeviceID                 : USB\ROOT_HUB30\4&5CDA812&0&0
PowerManagementCapabilities :
PowerManagementSupported    :
Present                     : True
Service                     : USBHUB3
Status                      : OK
StatusInfo                  :
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : LAPTOP-D47E8127
PSComputerName              : LAPTOP-D47E8127

__GENUS                     : 2
__CLASS                     : Win32_PnPEntity
__SUPERCLASS                : CIM_LogicalDevice
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_PnPEntity.DeviceID="ROOT\\USB\\0000"
__PROPERTY_COUNT            : 26
__DERIVATION                : {CIM_LogicalDevice, CIM_LogicalElement, CIM_ManagedSystemElement}
__SERVER                    : LAPTOP-D47E8127
__NAMESPACE                 : root\cimv2
__PATH                      : \\LAPTOP-D47E8127\root\cimv2:Win32_PnPEntity.DeviceID="ROOT\\USB\\0000"
Availability                :
Caption                     : DAEMON Tools Lite Virtual USB Bus
ClassGuid                   : {36fc9e60-c465-11cf-8056-444553540000}
CompatibleID                :
ConfigManagerErrorCode      : 0
ConfigManagerUserConfig     : False
CreationClassName           : Win32_PnPEntity
Description                 : DAEMON Tools Lite Virtual USB Bus
DeviceID                    : ROOT\USB\0000
ErrorCleared                : 
ErrorDescription            :
HardwareID                  : {root\dtliteusbbus}
InstallDate                 :
LastErrorCode               :
Manufacturer                : Disc Soft Ltd
Name                        : DAEMON Tools Lite Virtual USB Bus
PNPClass                    : USB
PNPDeviceID                 : ROOT\USB\0000
PowerManagementCapabilities :
PowerManagementSupported    :
Present                     : True
Service                     : dtliteusbbus
Status                      : OK
StatusInfo                  :
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : LAPTOP-D47E8127
PSComputerName              : LAPTOP-D47E8127

ServiceName      : RtlWlanu
MACAddress       : 28:EE:52:09:DB:A7
AdapterType      : Ethernet 802.3
DeviceID         : 2
Name             : TP-Link Wireless USB Adapter #4
NetworkAddresses :
Speed            : 144400000

__GENUS                     : 2
__CLASS                     : Win32_PnPEntity
__SUPERCLASS                : CIM_LogicalDevice
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_PnPEntity.DeviceID="USB\\VID_04F2&PID_B571&MI_00\\6&1EF97C42&0&0000"
__PROPERTY_COUNT            : 26
__DERIVATION                : {CIM_LogicalDevice, CIM_LogicalElement, CIM_ManagedSystemElement}
__SERVER                    : LAPTOP-D47E8127
__NAMESPACE                 : root\cimv2
__PATH                      : \\LAPTOP-D47E8127\root\cimv2:Win32_PnPEntity.DeviceID="USB\\VID_04F2&PID_B571&MI_00\\6&1EF97C42&0&0000"
Availability                :
Caption                     : HD WebCam
ClassGuid                   : {ca3e7ab9-b4c3-4ae6-8251-579ef933890f}
CompatibleID                : {USB\Class_0e&SubClass_03&Prot_00, USB\Class_0e&SubClass_03, USB\Class_0e}
ConfigManagerErrorCode      : 0
ConfigManagerUserConfig     : False
CreationClassName           : Win32_PnPEntity
Description                 : Périphérique vidéo USB
DeviceID                    : USB\VID_04F2&PID_B571&MI_00\6&1EF97C42&0&0000
ErrorCleared                :
ErrorDescription            :
HardwareID                  : {USB\VID_04F2&PID_B571&REV_9952&MI_00, USB\VID_04F2&PID_B571&MI_00}
InstallDate                 :
LastErrorCode               :
Manufacturer                : Microsoft
Name                        : HD WebCam
PNPClass                    : Camera
PNPDeviceID                 : USB\VID_04F2&PID_B571&MI_00\6&1EF97C42&0&0000
PowerManagementCapabilities :
PowerManagementSupported    :
Present                     : True
Service                     : usbvideo
Status                      : OK
StatusInfo                  :
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : LAPTOP-D47E8127
PSComputerName              : LAPTOP-D47E8127

__GENUS                     : 2
__CLASS                     : Win32_PnPEntity
__SUPERCLASS                : CIM_LogicalDevice
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_PnPEntity.DeviceID="USB\\VID_04F3&PID_9999&MI_00\\6&EED2AD0&0&12345"
__PROPERTY_COUNT            : 26
__DERIVATION                : {CIM_LogicalDevice, CIM_LogicalElement, CIM_ManagedSystemElement}
__SERVER                    : LAPTOP-D47E8127
__NAMESPACE                 : root\cimv2
__PATH                      : \\LAPTOP-D47E8127\root\cimv2:Win32_PnPEntity.DeviceID="USB\\VID_04F3&PID_9999&MI_00\\6&EED2AD0&0&12345"
Availability                :
Caption                     : Périphérique d’entrée USB
ClassGuid                   : {745a17a0-74d3-11d0-b6fe-00a0c90f57da}
CompatibleID                : {USB\Class_03&SubClass_01}
ConfigManagerErrorCode      : 0
ConfigManagerUserConfig     : False
CreationClassName           : Win32_PnPEntity
Description                 : Périphérique d’entrée USB
DeviceID                    : USB\VID_04F3&PID_9999&MI_00\6&EED2AD0&0&12345
ErrorCleared                :
ErrorDescription            :
HardwareID                  : {USB\Vid_04f3&Pid_9999&MI_00}
InstallDate                 :
LastErrorCode               :
Manufacturer                : (Périphériques système standard)
Name                        : Périphérique d’entrée USB
PNPClass                    : HIDClass
PNPDeviceID                 : USB\VID_04F3&PID_9999&MI_00\6&EED2AD0&0&12345
PowerManagementCapabilities :
PowerManagementSupported    :
Present                     : True
Service                     : HidUsb
Status                      : OK
StatusInfo                  :
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : LAPTOP-D47E8127
PSComputerName              : LAPTOP-D47E8127

__GENUS                     : 2
__CLASS                     : Win32_PointingDevice
__SUPERCLASS                : CIM_PointingDevice
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_PointingDevice.DeviceID="USB\\VID_04F3&PID_9999&MI_00\\6&EED2AD0&0&12345"
__PROPERTY_COUNT            : 33
__DERIVATION                : {CIM_PointingDevice, CIM_UserDevice, CIM_LogicalDevice, CIM_LogicalElement...}
__SERVER                    : LAPTOP-D47E8127
__NAMESPACE                 : root\cimv2
__PATH                      : \\LAPTOP-D47E8127\root\cimv2:Win32_PointingDevice.DeviceID="USB\\VID_04F3&PID_9999&MI_00\\6&EED2AD0&0&12345"
Availability                :
Caption                     : Périphérique d’entrée USB
ConfigManagerErrorCode      : 0
ConfigManagerUserConfig     : False
CreationClassName           : Win32_PointingDevice
Description                 : Périphérique d’entrée USB
DeviceID                    : USB\VID_04F3&PID_9999&MI_00\6&EED2AD0&0&12345
DeviceInterface             : 162
DoubleSpeedThreshold        :
ErrorCleared                :
ErrorDescription            :
Handedness                  :
HardwareType                : Périphérique d’entrée USB
InfFileName                 : input.inf
InfSection                  : HID_Inst.NT
InstallDate                 :
IsLocked                    :
LastErrorCode               :
Manufacturer                : (Périphériques système standard)
Name                        : Périphérique d’entrée USB
NumberOfButtons             : 0
PNPDeviceID                 : USB\VID_04F3&PID_9999&MI_00\6&EED2AD0&0&12345
PointingType                : 2
PowerManagementCapabilities :
PowerManagementSupported    : False
QuadSpeedThreshold          :
Resolution                  :
SampleRate                  :
Status                      : OK
StatusInfo                  :
Synch                       :
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : LAPTOP-D47E8127
PSComputerName              : LAPTOP-D47E8127

__GENUS                     : 2
__CLASS                     : Win32_USBController
__SUPERCLASS                : CIM_USBController
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_USBController.DeviceID="PCI\\VEN_8086&DEV_9D2F&SUBSYS_10941025&REV_21\\3&11583659&0&A0"
__PROPERTY_COUNT            : 23
__DERIVATION                : {CIM_USBController, CIM_Controller, CIM_LogicalDevice, CIM_LogicalElement...}
__SERVER                    : LAPTOP-D47E8127
__NAMESPACE                 : root\cimv2
__PATH                      : \\LAPTOP-D47E8127\root\cimv2:Win32_USBController.DeviceID="PCI\\VEN_8086&DEV_9D2F&SUBSYS_10941025&REV_21\\3&11583659&0&A0
                              "
Availability                :
Caption                     : Contrôleur hôte Intel(R) USB 3.0 eXtensible - 1.0 (Microsoft)
ConfigManagerErrorCode      : 0
ConfigManagerUserConfig     : False
CreationClassName           : Win32_USBController
Description                 : Contrôleur d’hôte compatible xHCI USB
DeviceID                    : PCI\VEN_8086&DEV_9D2F&SUBSYS_10941025&REV_21\3&11583659&0&A0
ErrorCleared                :
ErrorDescription            :
InstallDate                 : 
LastErrorCode               :
Manufacturer                : Contrôleur d’hôte xHCI USB générique
MaxNumberControlled         :
Name                        : Contrôleur hôte Intel(R) USB 3.0 eXtensible - 1.0 (Microsoft)
PNPDeviceID                 : PCI\VEN_8086&DEV_9D2F&SUBSYS_10941025&REV_21\3&11583659&0&A0
PowerManagementCapabilities :
PowerManagementSupported    :
ProtocolSupported           : 16
Status                      : OK
StatusInfo                  :
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : LAPTOP-D47E8127
TimeOfLastReset             :
PSComputerName              : LAPTOP-D47E8127

__GENUS                     : 2
__CLASS                     : Win32_USBController
__SUPERCLASS                : CIM_USBController
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_USBController.DeviceID="ROOT\\USB\\0000"
__PROPERTY_COUNT            : 23
__DERIVATION                : {CIM_USBController, CIM_Controller, CIM_LogicalDevice, CIM_LogicalElement...}
__SERVER                    : LAPTOP-D47E8127
__NAMESPACE                 : root\cimv2
__PATH                      : \\LAPTOP-D47E8127\root\cimv2:Win32_USBController.DeviceID="ROOT\\USB\\0000"
Availability                :
Caption                     : DAEMON Tools Lite Virtual USB Bus
ConfigManagerErrorCode      : 0
ConfigManagerUserConfig     : False
CreationClassName           : Win32_USBController
Description                 : DAEMON Tools Lite Virtual USB Bus
DeviceID                    : ROOT\USB\0000
ErrorCleared                :
ErrorDescription            :
InstallDate                 :
LastErrorCode               :
Manufacturer                : Disc Soft Ltd
MaxNumberControlled         :
Name                        : DAEMON Tools Lite Virtual USB Bus
PNPDeviceID                 : ROOT\USB\0000
PowerManagementCapabilities :
PowerManagementSupported    :
ProtocolSupported           : 16
Status                      : OK
StatusInfo                  :
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : LAPTOP-D47E8127
TimeOfLastReset             :
PSComputerName              : LAPTOP-D47E8127

__GENUS                     : 2
__CLASS                     : Win32_USBHub
__SUPERCLASS                : CIM_USBHub
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_USBHub.DeviceID="USB\\VID_04F2&PID_B571\\5&651EB89&0&7"
__PROPERTY_COUNT            : 28
__DERIVATION                : {CIM_USBHub, CIM_USBDevice, CIM_LogicalDevice, CIM_LogicalElement...}
__SERVER                    : LAPTOP-D47E8127
__NAMESPACE                 : root\cimv2
__PATH                      : \\LAPTOP-D47E8127\root\cimv2:Win32_USBHub.DeviceID="USB\\VID_04F2&PID_B571\\5&651EB89&0&7"
Availability                :
Caption                     : Périphérique USB composite
ClassCode                   :
ConfigManagerErrorCode      : 0
ConfigManagerUserConfig     : False
CreationClassName           : Win32_USBHub
CurrentAlternateSettings    :
CurrentConfigValue          :
Description                 : Périphérique USB composite
DeviceID                    : USB\VID_04F2&PID_B571\5&651EB89&0&7
ErrorCleared                :
ErrorDescription            :
GangSwitched                :
InstallDate                 :
LastErrorCode               :
Name                        : Périphérique USB composite
NumberOfConfigs             :
NumberOfPorts               :
PNPDeviceID                 : USB\VID_04F2&PID_B571\5&651EB89&0&7
PowerManagementCapabilities :
PowerManagementSupported    :
ProtocolCode                :
Status                      : OK
StatusInfo                  :
SubclassCode                :
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : LAPTOP-D47E8127
USBVersion                  :
PSComputerName              : LAPTOP-D47E8127

__GENUS                     : 2
__CLASS                     : Win32_USBHub
__SUPERCLASS                : CIM_USBHub
__DYNASTY                   : CIM_ManagedSystemElement
__RELPATH                   : Win32_USBHub.DeviceID="USB\\ROOT_HUB30\\4&5CDA812&0&0"
__PROPERTY_COUNT            : 28
__DERIVATION                : {CIM_USBHub, CIM_USBDevice, CIM_LogicalDevice, CIM_LogicalElement...}
__SERVER                    : LAPTOP-D47E8127
__NAMESPACE                 : root\cimv2
__PATH                      : \\LAPTOP-D47E8127\root\cimv2:Win32_USBHub.DeviceID="USB\\ROOT_HUB30\\4&5CDA812&0&0"
Availability                :
Caption                     : Hub USB racine (USB 3.0)
ClassCode                   :
ConfigManagerErrorCode      : 0
ConfigManagerUserConfig     : False
CreationClassName           : Win32_USBHub
CurrentAlternateSettings    :
CurrentConfigValue          :
Description                 : Hub USB racine (USB 3.0)
DeviceID                    : USB\ROOT_HUB30\4&5CDA812&0&0
ErrorCleared                :
ErrorDescription            :
GangSwitched                :
InstallDate                 :
LastErrorCode               :
Name                        : Hub USB racine (USB 3.0)
NumberOfConfigs             :
NumberOfPorts               :
PNPDeviceID                 : USB\ROOT_HUB30\4&5CDA812&0&0
PowerManagementCapabilities :
PowerManagementSupported    :
ProtocolCode                :
Status                      : OK
StatusInfo                  :
SubclassCode                :
SystemCreationClassName     : Win32_ComputerSystem
SystemName                  : LAPTOP-D47E8127
USBVersion                  :
PSComputerName              : LAPTOP-D47E8127
```

**Comment by Zoliex** [comment]
I have corrected the command for windows in your plugin (I just removed the ^) and now i get my USB devices : 
```json
[
  {
    bus: null,
    deviceId: 'USB\\VID_04F3&PID_9999&MI_00\\6&EED2AD0&0&12345',
    id: 9,
    name: 'Périphérique d’entrée USB',
    type: 'Mouse',
    removable: null,
    vendor: null,
    manufacturer: '(Périphériques système standard)',
    maxPower: null,
    serialNumber: null
  },
  {
    bus: null,
    deviceId: 'PCI\\VEN_8086&DEV_9D2F&SUBSYS_10941025&REV_21\\3&11583659&0&A0',
    id: 10,
    name: 'Contrôleur hôte Intel(R) USB 3.0 eXtensible - 1.0 (Microsoft)',
    type: 'Controller',
    removable: null,
    vendor: null,
    manufacturer: 'Contrôleur d’hôte xHCI USB générique',
    maxPower: null,
    serialNumber: null
  },
  {
    bus: null,
    deviceId: 'ROOT\\USB\\0000',
    id: 11,
    name: 'DAEMON Tools Lite Virtual USB Bus',
    type: 'Controller',
    removable: null,
    vendor: null,
    manufacturer: 'Disc Soft Ltd',
    maxPower: null,
    serialNumber: null
  },
  {
    bus: null,
    deviceId: 'USB\\VID_04F2&PID_B571\\5&651EB89&0&7',
    id: 12,
    name: 'Périphérique USB composite',
    type: 'Hub',
    removable: null,
    vendor: null,
    manufacturer: '',
    maxPower: null,
    serialNumber: null
  },
  {
    bus: null,
    deviceId: 'USB\\ROOT_HUB30\\4&5CDA812&0&0',
    id: 13,
    name: 'Hub USB racine (USB 3.0)',
    type: 'Hub',
    removable: null,
    vendor: null,
    manufacturer: '',
    maxPower: null,
    serialNumber: null
  }
]
```

**Comment by sebhildebrandt** [comment]
@Zoliex This makes sense! Will publish a new release in a few minutes.

**Comment by Zoliex** [comment]
OK good luck 🍀
Have a good day

**Comment by sebhildebrandt** [comment]
@Zoliex I just released a new version 5.10.6 that should fix the problem.

**Comment by Zoliex** [comment]
Thanks for the fix
Have a good day
